jiant/tasks/seq2seq.py

```diff
@@ -118,10 +119,22 @@ def get_metrics(self, reset=False):
         acc = self.scorer2.get_metric(reset)
         return {"perplexity": math.exp(avg_nll), "accuracy": acc}
 
-    def update_metrics(self, logits, labels, tagmask=None):
-        # Cut logits, since generation is done until max seq length (usually >> target length).
-        relevant_logits = logits.max(dim=2)[1][:, : labels.shape[1]]
-        self.scorer2(relevant_logits, labels, tagmask)
+    def update_metrics(self, logits, labels, tagmask=None, predictions=None):
+        assert logits is not None or predictions is not None
+
+        if logits is not None:
+            # Cut logits, since generation is done until max seq length (usually >> target length).
+            relevant_logits = logits.max(dim=2)[1][:, : labels.shape[1]]
+            self.scorer2(relevant_logits, labels, tagmask)
+        else:
```

Comment:
quick comment: `# logits = None, predictions is not None`

Reply:
Done.